crates/biome_css_analyze/src/lint/nursery/no_font_family_duplicate_names.rs

```diff
@@ -0,0 +1,200 @@
+use std::collections::HashSet;
+
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    AnyCssGenericComponentValue, AnyCssValue, CssGenericComponentValueList, CssGenericProperty,
+};
+use biome_rowan::{AstNode, SyntaxNodeCast, TextRange};
+
+use crate::utils::{is_css_variable, is_font_family_keyword, is_font_shorthand_keyword};
+
+declare_rule! {
+    /// Disallow duplicate names within font families.
+    ///
+    /// This rule checks the `font` and `font-family` properties for duplicate font names.
+    ///
+    /// This rule ignores var(--custom-property) variable syntaxes.
```

Comment:
This is actually a good excuse to evaluate the implementation of a possible semantic model for CSS :) 

Of course, not in this PR!

Reply:
Yeah, it will come true in the near future:) https://github.com/biomejs/biome/pull/2308/commits/e8510f46418b72f3c28e53f37408335f023a6636